### PR TITLE
Add new default values for commands ran from XCode

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What and why?

We can now run the `xcode` command with `../node_modules/.bin/datadog-ci react-native xcode`, instead of `yarn datadog-ci react-native xcode`.
This changes the current working directory to be the `ios/` folder of the RN app instead of the root of the RN app.
We have a similar change for the dsyms command.

The default value for the `react-native` path should be in this case `../node_modules/react-native` instead of `node_modules/react-native`.
Similarly, the default path for the `datadog-ci.json` file should be `../datadog-ci.json` instead of `./datadog-ci.json`.

### How?

I've left the legacy values to be picked up first to avoid regressions.
I've only changed the default path for `datadog-ci.json` for the react native and dsyms upload commands.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Review by 0xnm — APPROVED** [review]

File: src/commands/flutter-symbols/upload.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ export class UploadCommand extends Command {
 
     this.config = await resolveConfigFromFile(this.config, {
       configPath: this.configPath,
-      defaultConfigPath: DEFAULT_CONFIG_PATH,
+      defaultConfigPaths: ['datadog-ci.json'],
```
Comment (fuzzybinary) [inline src/commands/flutter-symbols/upload.ts:122]: Why not use `DEFAULT_CONFIG_PATHS` here?

**Review by fuzzybinary — APPROVED** [review]

File: src/commands/flutter-symbols/upload.ts (hunk 1 above)
Comment (louiszawadzki) [reply src/commands/flutter-symbols/upload.ts:122]: I don't want to create any regression on other commands - if you're fine with `'datadog-ci.json'`, having `'../datadog-ci.json'` will only create regressions.
If you think that `DEFAULT_CONFIG_PATHS` should rather be `['datadog-ci.json']`, that also makes sense to me and I can change it

**Review by cswatt — APPROVED** [review]
👍 for docs

File: src/commands/flutter-symbols/upload.ts (hunk 1 above)
Comment (lefebvree) [reply src/commands/flutter-symbols/upload.ts:122]: I think it'd makes sense to keep `DEFAULT_CONFIG_PATHS = ['datadog-ci.json']`, use it everywhere and have `react-native` be the exception with `['datadog-ci.json', '../datadog-ci.json']`

**Review by lefebvree — APPROVED** [review]

**Review by duncanista — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**